src/components/pages/identity-overview/identity-overview.tsx

```diff
@@ -14,64 +15,87 @@ import { DeleteConfirmationBoxContent } from "@src/components/confirmation-box/d
 import { PrimaryConfirmationBoxContent } from "@src/components/confirmation-box/primary-confirmation-box-content";
 import { NeutralLink } from "@src/components/neutral-link/neutral-link";
 import { SkeletonTextLine } from "@src/components/skeletons/skeletons";
+import { fetchJson } from "@src/utils/fetch-json";
 import { getIdentityType } from "@src/utils/get-identity-type";
 
 const IdentityOverview: React.FC<{
-  data?: { identity: Identity };
-}> = ({ data }) => {
+  identity?: Identity;
+}> = ({ identity }) => {
   const [confirmationBoxOpen, setConfirmationBoxOpen] =
     React.useState<boolean>(false);
   const [preventAnimation, setPreventAnimation] = React.useState<boolean>(true);
   const [confirmationBoxContent, setConfirmationBoxContent] =
     React.useState<JSX.Element>(<React.Fragment />);
+  const router = useRouter();
+
+  const deleteIdentity = (identity: Identity): Promise<void> => {
+    const requestData = {
+      type: getIdentityType(identity.type),
+      value: identity.value,
+      id: identity.id,
+    };
+
+    return fetchJson(
+      `/api/identities/${identity.id}`,
+      "DELETE",
+      requestData,
+    ).then(() => {
+      router && router.push("/account/logins");
+    });
+  };
+
+  const markIdentityAsPrimary = (identity: Identity): Promise<void> =>
+    fetchJson(`/api/identities/${identity.id}/mark-as-primary`, "POST", {
+      identityId: identity.id,
+    }).then(() => {
+      router && router.push("/account/logins");
+    });
```

Comment:
Those two functions replace what we previously had inside the `ConfirmationBoxContent` components.
I think this component is the right place to define such functions.